src/ui/controls/edit.cpp

```diff
@@ -2303,12 +2306,15 @@ void CEdit::MoveChar(int move, bool bWord, bool bSelect)
         }
         else
         {
-            m_cursor1 --;
-            if ( m_cursor1 < 0 )  m_cursor1 = 0;
+            if ( m_cursor1 > 0 )
+            {
+                m_cursor1 --;
+                while ( m_cursor1 > 0 && (m_text[m_cursor1] & 0xC0) == 0x80 )  m_cursor1 --;
```

Comment:
This `m_text[m_cursor1] & 0xC0) == 0x80` condition repeats a few times, maybe extract it to a method/function with a readable name. Not knowing UTF-8, I have no idea what's happening here just by reading this code.

Reply:
This is not entirely fair, because the code already contains hex values.

Reply:
And how to name this function?

Reply:
What? I don't understand what you want to say here.

I literally see exactly the same code in 4 places just in this PR, there's probably more places in the whole file. Only the `m_text[m_cursor1]` part changes which can be easily passed as parameter. It's breaking the (quite fundamental) DRY rule.

Reply:
If strictly adhere to DRY. Ok.

Reply:
Regarding the name of the function, maybe `isUtf8ContinuationByte`?

```cpp
bool isUtf8ContinuationByte(char c)
{
    return (c & 0b1100'0000) == 0b1000'0000;
}
```

Now the line would look like this:

```cpp
while ( m_cursor1 > 0 && isUtf8ContinuationByte(m_text[m_cursor1]) ) m_cursor1 --;
```

This is much more readable IMHO, it is now obvious what the line does, assuming you know what is a continuation byte.

The function body also better describes itself thanks to the binary literals. Continuation byte is defined as a byte where the top two bits are `10`. So, firstly, since we only care about the top two bits, we do `c & 0b1100'0000` to zero the irrelevant bits, and then check if the two bits are indeed `10`.

Reply:
isUtf8ContinuationByte function is good idea. I will add it.